src/SqlConverters.cs

```diff
@@ -188,10 +191,12 @@ public virtual async Task<string> BuildItemFromAttributeAsync(SqlAttribute attri
 
             IAsyncEnumerable<T> IConverter<SqlAttribute, IAsyncEnumerable<T>>.Convert(SqlAttribute attribute)
             {
-                TelemetryInstance.TrackConvert(ConvertType.IAsyncEnumerable);
                 try
                 {
-                    return new SqlAsyncEnumerable<T>(SqlBindingUtilities.BuildConnection(attribute.ConnectionStringSetting, this._configuration), attribute);
+                    SqlConnection conn = SqlBindingUtilities.BuildConnection(attribute.ConnectionStringSetting, this._configuration);
+                    Dictionary<TelemetryPropertyName, string> props = conn.AsConnectionProps();
```

Comment:
You're trying to access the properties here before the connection is opened which is likely why the tests are failing now.

Reply:
Actually, for that second part I'll make a new issue and we can follow up with that later. No reason to clutter up this PR even more.

Reply:
So I can't really collect the server version unless the connection is open?

Reply:
https://github.com/Azure/azure-functions-sql-extension/issues/338

Reply:
Given that you need to talk to the server to know what version it is, yes that would be extremely difficult to do without an open connection 😄

Reply:
But am collecting that info prior to opening the connection in BuildItemFromAttributeAsync, how doeds that work?